File: Python/Warmup-2/array123.py
"""
Given an array of ints, return True if the sequence of numbers 1, 2, 3 appears in the array somewhere.

array123([1, 1, 2, 3, 1]) → True
array123([1, 1, 2, 4, 1]) → False
array123([1, 1, 2, 1, 2, 3]) → True
"""

def array123(nums):
  for i in range(len(nums)):
    if nums[i]==1:
      if i+1<len(nums) and nums[i+1]==2:
        if i+2<len(nums) and nums[i+2]==3:
          return True
  return False

File: Java/Warmup-2/array123.java
/*
Given an array of ints, return true if the sequence of numbers 1, 2, 3 appears in the array somewhere.

array123([1, 1, 2, 3, 1]) ? true
array123([1, 1, 2, 4, 1]) ? false
array123([1, 1, 2, 1, 2, 3]) ? true
*/

public boolean array123(int[] nums) {
  for(int i=0;i<nums.length;i++){
    if(nums[i]==1 && i+1<nums.length){
      if(nums[i+1]==2 && i+2<nums.length){
        if(nums[i+2]==3){
          return true;
        }
      }
    }
  }
  return false;
}